nova_vm/src/ecmascript/types/language/bigint.rs

```diff
@@ -278,6 +387,53 @@ impl BigInt {
         }
     }
 
+    /// ### [6.1.6.2.6 BigInt::remainder ( n, d )](https://tc39.es/ecma262/#sec-numeric-types-bigint-remainder)
+    pub(crate) fn remainder(agent: &mut Agent, n: BigInt, d: BigInt) -> JsResult<BigInt> {
+        match (n, d) {
+            (BigInt::SmallBigInt(n), BigInt::SmallBigInt(d)) => {
+                if d == SmallBigInt::zero() {
+                    return Err(agent.throw_exception_with_static_message(
+                        ExceptionType::RangeError,
+                        "Division by zero",
+                    ));
+                }
+                let (n, d) = (n.into_i64() as i128, d.into_i64() as i128);
+                let result = n % d;
```

Comment:
**needs (nonblocking):** The result of the remainder will always have smaller absolute value than the denominator, so the there's not only no chance of overflowing into `i128`, but the result should always be a `SmallBigInt`.